yt_dlp/extractor/mx3.py

```diff
@@ -0,0 +1,98 @@
+import re
+
+from .common import InfoExtractor
+from ..utils import urlhandle_detect_ext
+from ..networking import HEADRequest
+
+
+class Mx3IE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?mx3\.ch/t/(?P<id>[0-9A-Za-z]+)'
```

Comment:
There also exists https://neo.mx3.ch/ and https://volksmusik.mx3.ch/
I think the extractor will also need to be slightly adjusted to work with both.

Reply:
Ah I didn't explore those until now - they have quite some additional content. I've added them both.

Reply:
You are now subclassing from a concrete IE, which should not be done.
Instead, I would add another capture group for the subdomain and use that for the api calls.

Reply:
Capture group sounds like an elegant solution, but this means the 'id' needs to contain the domain too, because it seems like the sites do not share a common id namespace (see my commit messages). That's probably okay? Otherwise I'll just do an abstract base class.

Reply:
Ah right, forgot about that. Yeah, then it's better to move everything to an abstract base class.